### PR TITLE
Allow template variables to be cancelled

### PR DESCRIPTION
**Pull request by kaydelaney**
Allows the loading indicator to be clicked in order to cancel a loading template variable.

Follow up to https://github.com/grafana/scenes/pull/220

### CODE REVIEW

File: packages/scenes/src/variables/variants/query/QueryVariable.tsx [hunk 1]
```diff
@@ -100,6 +102,12 @@ export class QueryVariable extends MultiValueVariable<QueryVariableState> {
     );
   }
 
+  public cancel() {
+    this.setState({ loading: false });
+    const sceneVarSet = getClosest(this, (s) => s instanceof SceneVariableSet ? s : undefined);
+    sceneVarSet?.cancel(this);
+  }
```
Comment (torkelo) [inline packages/scenes/src/variables/variants/query/QueryVariable.tsx:109]: think you can move this cancel function implementation to the abstract class

File: packages/scenes/src/variables/variants/query/QueryVariable.tsx [hunk 2]
```diff
@@ -100,6 +102,12 @@ export class QueryVariable extends MultiValueVariable<QueryVariableState> {
     );
   }
 
+  public cancel() {
+    this.setState({ loading: false });
+    const sceneVarSet = getClosest(this, (s) => s instanceof SceneVariableSet ? s : undefined);
```
Comment (torkelo) [inline packages/scenes/src/variables/variants/query/QueryVariable.tsx:107]:
```suggestion
    const sceneVarSet = this.parent as SceneVariableSet;
```

We assume variables are inside variable sets (at least for now) :)

**Review by torkelo — COMMENTED** [review]
Looks good, tricky typing issue with the onCancel method. 

Only other option I can see is wrapping the Select in a React context that contains the variable, and accessing the context from the LoadingIndicator.

File: packages/scenes/src/variables/variants/query/QueryVariable.tsx (hunk 1 above)
Comment (kaydelaney) [reply packages/scenes/src/variables/variants/query/QueryVariable.tsx:109]: Hmm, I get some type errors related to
```typescript
this.setState({ loading: false });
```
if I move it to the abstract class:
"Argument of type '{ loading: false; }' is not assignable to parameter of type 'Partial\<TState\>'. ts(2345)"

Comment (torkelo) [reply packages/scenes/src/variables/variants/query/QueryVariable.tsx]: @kaydelaney  try using setStateHelper (function on MultiValueVariable that works around a typescript issue)

**Review by dprokop — APPROVED** [review]
Looks and works well. One thing missing is some tests of the cancellation flow, think it would be worth adding those to make sure that cancellation is propagated correctly.

**Pull request merged**